OOM errors from Electron child process when running screenshot tests

I'm consistently getting out-of-memory errors when trying to run screenshot tests with `cypress-image-snapshot`. 

The stack trace is quite busy, but it _looks_ like something deep within this library's actions on `after:screenshot` is causing this...

It should be noted that I'm diffing full-page screenshots, which weigh in at 1 and 3.4MB, though this error is happening with the 1MB test case and was not happening when I only had the 3.4MB test case. It's possible that I'm plainly pushing Cypress or this library too hard 😄 

I'm doing this in an open-source project, so you can see my entire setup here: https://github.com/swagger-api/swagger-ui/compare/regression/schemes-servers-rendering?expand=1

---

Output:

```
  Running: screenshot/openapi3-petstore-expanded.js...                                     (5 of 6)


  OpenAPI 3 Petstore Expanded Screenshots

<--- Last few GCs --->

[16880:0x7ffb58032e00]    69441 ms: Scavenge 11.8 (17.0) -> 10.8 (17.0) MB, 0.0 / 0.0 ms  allocation failure
[16880:0x7ffb58032e00]    69445 ms: Scavenge 11.8 (17.0) -> 10.8 (17.0) MB, 0.0 / 0.0 ms  allocation failure
[16880:0x7ffb58032e00]    69852 ms: Mark-sweep 139.2 (144.6) -> 136.6 (143.1) MB, 0.7 / 0.0 ms  (+ 6.5 ms in 7 steps since start of marking, biggest step 2.1 ms, walltime since start of marking 406 ms) finalize incremental marking via stack guard GC in ol

<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x1d50276ad681 <JSObject>
    1: toJSON [buffer.js:~931] [pc=0x3c6d7b026bdd](this=0x473c10823b9 <Uint8Array map = 0x224d9534ffc1>)
    2: arguments adaptor frame: 1->0
    5: builtin exit frame: stringify(this=0x1d5027693ec9 <JSON map = 0x224d95306249>,0x29d5fd282311 <undefined>,0x473c1083cc1 <JSFunction JSON.stringify (sfi = 0x3f0a3617dd19)>,0x473c1083c99 <Object map = 0x2f41d02c34f1>)

    6: arguments adaptor frame: 2...

FATAL ERROR: invalid table size Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 2: node::FatalError(char const*, char const*) [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 3: v8::internal::FatalProcessOutOfMemory(char const*) [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 4: v8::internal::FatalProcessOutOfMemory(char const*) [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 5: v8::internal::JSArrayBuffer::Setup(v8::internal::Handle<v8::internal::JSArrayBuffer>, v8::internal::Isolate*, bool, void*, unsigned long, v8::internal::SharedFlag) [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 6: v8::internal::StringTable::LookupString(v8::internal::Isolate*, v8::internal::Handle<v8::internal::String>) [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 7: v8::internal::EhFrameIterator::DecodeSLeb128(unsigned char const*, int*) [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 8: v8::internal::HeapNumber::HeapNumberPrint(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
 9: v8::internal::RegisterConfiguration::AreAliases(v8::internal::MachineRepresentation, int, v8::internal::MachineRepresentation, int) const [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
10: v8::internal::OldSpaces::next() [/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib]
11: 0x3c6d7ae843fd

Error: The following error was thrown by a plugin. We've stopped running your tests because a plugin crashed.

Error [ERR_IPC_CHANNEL_CLOSED]: channel closed
    at ChildProcess.target.send (internal/child_process.js:584:16)
    at Object.send (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:34:27)
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:74:28
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:77:16
    at Promise.cancellationExecute [as _execute] (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/debuggability.js:321:9)
    at Promise._resolveFromExecutor (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:77:14)
    at Object.wrapParentPromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:61:14)
    at Object.after:screenshot (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:67:27)
    at Object.execute (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:115:38)
    at Object.afterScreenshot (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/screenshots.js:411:22)
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/screenshot.js:16:32
    at tryCatcher (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:431:57)
    at Promise._settlePromiseFromHandler (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:522:17)
    at Promise._settlePromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:431:57)
    at Promise._settlePromiseFromHandler (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:522:17)
    at Promise._settlePromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:636:18)
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/nodeback.js:42:21
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqWrap.oncomplete (fs.js:153:5)

    at Object.get (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/errors.js:220:11)
    at ChildProcess.handleError (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:92:24)
    at emitOne (events.js:115:13)
    at ChildProcess.emit (events.js:210:7)
    at ChildProcess.target.send (internal/child_process.js:588:12)
    at Object.send (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:34:27)
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:74:28
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:77:16
    at Promise.cancellationExecute [as _execute] (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/debuggability.js:321:9)
    at Promise._resolveFromExecutor (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:77:14)
    at Object.wrapParentPromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:61:14)
    at Object.after:screenshot (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:67:27)
    at Object.execute (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:115:38)
    at Object.afterScreenshot (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/screenshots.js:411:22)
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/screenshot.js:16:32
    at tryCatcher (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:431:57)
    at Promise._settlePromiseFromHandler (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:522:17)
    at Promise._settlePromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:431:57)
    at Promise._settlePromiseFromHandler (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:522:17)
    at Promise._settlePromise (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:636:18)
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/nodeback.js:42:21
    at /Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqWrap.oncomplete (fs.js:153:5)


  (Results)

  ┌────────────────────────────────────────────────────────┐
  │ Tests:        0                                        │
  │ Passing:      0                                        │
  │ Failing:      1                                        │
  │ Pending:      0                                        │
  │ Skipped:      0                                        │
  │ Screenshots:  0                                        │
  │ Video:        false                                    │
  │ Duration:     0 seconds                                │
  │ Spec Ran:     screenshot/openapi3-petstore-expanded.js │
  └────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running: screenshot/swagger2-petstore.js...                                              (6 of 6)

Error: The following error was thrown by a plugin. We've stopped running your tests because a plugin crashed.

Error [ERR_IPC_CHANNEL_CLOSED]: channel closed
    at ChildProcess.target.send (internal/child_process.js:584:16)
    at Object.send (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:34:27)
    at EventEmitter.<anonymous> (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/preprocessor.js:60:18)
    at emitOne (events.js:120:20)
    at EventEmitter.emit (events.js:210:7)
    at Object.removeFile (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/preprocessor.js:108:19)
    at Object.options.onBrowserClose (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/open_project.js:98:30)
    at EventEmitter.<anonymous> (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/browsers/index.js:111:21)
    at Object.onceWrapper (events.js:314:30)
    at emitNone (events.js:110:20)
    at EventEmitter.emit (events.js:207:7)
    at BrowserWindow.<anonymous> (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/browsers/electron.js:247:29)
    at Object.onceWrapper (events.js:316:30)
    at emitOne (events.js:120:20)
    at BrowserWindow.emit (events.js:210:7)

    at Object.get (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/errors.js:220:11)
    at ChildProcess.handleError (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:92:24)
    at emitOne (events.js:115:13)
    at ChildProcess.emit (events.js:210:7)
    at ChildProcess.target.send (internal/child_process.js:588:12)
    at Object.send (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:34:27)
    at EventEmitter.<anonymous> (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/preprocessor.js:60:18)
    at emitOne (events.js:120:20)
    at EventEmitter.emit (events.js:210:7)
    at Object.removeFile (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/preprocessor.js:108:19)
    at Object.options.onBrowserClose (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/open_project.js:98:30)
    at EventEmitter.<anonymous> (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/browsers/index.js:111:21)
    at Object.onceWrapper (events.js:314:30)
    at emitNone (events.js:110:20)
    at EventEmitter.emit (events.js:207:7)
    at BrowserWindow.<anonymous> (/Users/kyle/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/browsers/electron.js:247:29)
    at Object.onceWrapper (events.js:316:30)
    at emitOne (events.js:120:20)
    at BrowserWindow.emit (events.js:210:7)


  (Results)

  ┌───────────────────────────────────────────────┐
  │ Tests:        0                               │
  │ Passing:      0                               │
  │ Failing:      1                               │
  │ Pending:      0                               │
  │ Skipped:      0                               │
  │ Screenshots:  0                               │
  │ Video:        false                           │
  │ Duration:     0 seconds                       │
  │ Spec Ran:     screenshot/swagger2-petstore.js │
  └───────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


      Spec                                                Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔ bugs/4838.js                              00:13        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔ bugs/4867.js                              00:02        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔ bugs/editor-1868.js                       00:03        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔ deep-linking.js                           00:18       20       20        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖ screenshot/openapi3-petstore-expande…       0ms        -        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖ screenshot/swagger2-petstore.js             0ms        -        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    2 of 6 failed (33%)                         00:37       23       23        2        -        -

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! swagger-ui@3.19.1 test-e2e-cypress: `cypress run`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the swagger-ui@3.19.1 test-e2e-cypress script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kyle/.npm/_logs/2018-10-02T01_16_23_812Z-debug.log
ERROR: "test-e2e-cypress" exited with 2.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! swagger-ui@3.19.1 e2e-cypress: `run-p -r hot-e2e-cypress-server mock-api test-e2e-cypress`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the swagger-ui@3.19.1 e2e-cypress script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kyle/.npm/_logs/2018-10-02T01_16_23_967Z-debug.log
```

--- a/__tests__/command.test.js
+++ b/__tests__/command.test.js
@@ -7,6 +7,7 @@
 
 global.Cypress = {
   env: () => false,
+  config: () => '/cypress/screenshots',
   Commands: {
     add: jest.fn(),
   },
@@ -41,6 +42,7 @@ describe('command', () => {
     boundMatchImageSnapshot(subject, commandOptions);
 
     expect(cy.task).toHaveBeenCalledWith('Matching image snapshot', {
+      screenshotsFolder: '/cypress/screenshots',
       updateSnapshots: false,
       options: {
         failureThreshold: 10,

--- a/__tests__/plugin.test.js
+++ b/__tests__/plugin.test.js
@@ -29,6 +29,7 @@ describe('plugin', () => {
     process.cwd = () => '';
 
     const options = {
+      screenshotsFolder: '/cypress/screenshots',
       updateSnapshots: true,
     };
 

--- a/src/command.js
+++ b/src/command.js
@@ -7,6 +7,7 @@
 
 import { MATCH, RECORD } from './constants';
 
+const screenshotsFolder = Cypress.config('screenshotsFolder');
 const updateSnapshots = Cypress.env('updateSnapshots') || false;
 
 export function matchImageSnapshotCommand(defaultOptions) {
@@ -17,6 +18,7 @@ export function matchImageSnapshotCommand(defaultOptions) {
     };
 
     cy.task(MATCH, {
+      screenshotsFolder,
       updateSnapshots,
       options,
     });

--- a/src/plugin.js
+++ b/src/plugin.js
@@ -34,11 +34,12 @@ export function matchImageSnapshotPlugin({ path: screenshotPath }) {
   }
 
   const {
+    screenshotsFolder,
     updateSnapshots,
     options: {
       failureThreshold = 0,
       failureThresholdType = 'pixel',
-      customSnapshotsDir = '/cypress/snapshots',
+      customSnapshotsDir,
       customDiffDir,
       ...options
     } = {},
@@ -51,11 +52,9 @@ export function matchImageSnapshotPlugin({ path: screenshotPath }) {
   const screenshotDir = screenshotPath.replace(screenshotFileName, '');
   const relativePath = screenshotDir.match(/screenshots(.*)/)[1];
   const snapshotIdentifier = screenshotFileName.replace('.png', '');
-  const snapshotsDir = path.join(
-    process.cwd(),
-    customSnapshotsDir,
-    relativePath
-  );
+  const snapshotsDir = customSnapshotsDir
+    ? path.join(process.cwd(), customSnapshotsDir, relativePath)
+    : path.join(screenshotsFolder, '..', 'snapshots', relativePath);
 
   const snapshotKebabPath = path.join(
     snapshotsDir,

